### PR TITLE
break types into own files

### PR DESCRIPTION
**Pull request by gregra81**
break types into own files so that types are manageable as they grow

### CODE REVIEW

File: types/client-data.interface.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+type SubscribableEvents = 'context' | 'settings' | 'itemIds' | 'events';
+
+interface GetResponse {
+    value: any;
+    version: any;
+}
+
+interface SetResponse {
+    success: boolean;
+    reason?: string | undefined;
```
Comment (ShayElkana) [inline types/client-data.interface.ts:10]: Why optional & undefined?

**Comment by arutkowski00** [comment]
All definition types should end with `.d.ts` extension

File: types/client-api.interface.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+interface APIOptions {
```
Comment (arutkowski00) [inline types/client-api.interface.ts:1]: All type definition files should end with `.d.ts` so it is clear that they are not a compilable source code.

**Review by arutkowski00 — CHANGES_REQUESTED** [review]

File: types/client-api.interface.ts (hunk 2 above)
Comment (gregra81) [reply types/client-api.interface.ts:1]: @arutkowski00 initially I wanted to do it but these types are not exported and are only used for imports in the main index.d.ts file. Do you think it still requires have to be `.d.ts` files?

**Pull request merged**